openedx/core/lib/block_structure/block_structure.py

```diff
@@ -564,7 +564,7 @@ def remove_transformer_block_field(self, usage_key, transformer, key):
         """
         try:
             transformer_block_data = self.get_transformer_block_data(usage_key, transformer)
-            delattr(transformer_block_data, key)
+            del transformer_block_data.fields[key]
```

Comment:
This is exactly why the try-except wrapper is there - to catch the issue.  So what problem are you running into?